CONTRIBUTING.md

```diff
@@ -51,5 +51,104 @@ This process can be performed automatically by running `make license-generator`.
 
 Sometimes pull requests take a little while to be merged. This is partially because they often have knock-on effects for the rest of the ecosystem, and partially because we want to give core team members time to review and consider changes thoroughly. Please see the organization's [governance document](https://github.com/purescript/governance) for information about when a pull request may be merged.
 
+## Developer Guide
+
+The following instructions are intended to help PureScript users more easily contribute to the compiler, even if this is your first Haskell project.
+
+### Prerequisites
+
+Install `stack`. [Instructions](https://docs.haskellstack.org/en/stable/README/).
+
+See the [PureScript Install Guide](INSTALL.md) for information on the `curses`/`ncurses` dependency.
```

Comment:
As of v0.14.2, this is no longer relevant. Can we drop this? Anyone contributing to the repo won't need to be concerned about this anymore.